### PR TITLE
Rename bridges and make part of baremetal platform config

### LINKED ISSUE #70: Remove openstack terminology
The bootstrap VM's terraform config refers to the "overcloud" bridge. IMHO this should be renamed to something less attached to OpenStack tripleo.

```hcl
variable "overcloud_bridge" {
  type        = "string"
  description = "The name of the overcloud bridge"
}
```

### PR DESCRIPTION
**Pull request by stbenjam**
fixes #70

### CODE REVIEW

**Comment by stbenjam** [comment]
The values for the vm_bridge and baremetal_bridge are hardcoded: 

https://github.com/openshift-metal3/kni-installer/blob/master/pkg/asset/cluster/tfvars.go#L187-L188

Should the user be able to customize these? I imagine it could be part of the baremetal platform's install-config.

**Comment by hardys** [comment]
> Should the user be able to customize these? I imagine it could be part of the baremetal platform's install-config.

I guess we could add them to the installconfig but default to the existing values?

**Comment by hardys** [comment]
Looks good, but if we're going to expose these variables to the user via the installconfig, I wonder if the names could be more self-documenting?  E.g vm_bridge and baremetal_bridge doesn't really describe their purpose, since both are connected to the bootstrap VM, one for external access and one for provisioning?

**Comment by stbenjam** [comment]
Any preference? Maybe `provisioning_bridge`, and `external_bridge`?

**Comment by hardys** [comment]
> Any preference? Maybe `provisioning_bridge`, and `external_bridge`?

Sounds good to me :+1:

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/665/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/671/

**Comment by stbenjam** [comment]
@hardys Updated, and CI is passing

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/679/

File: pkg/asset/installconfig/baremetal/baremetal.go [hunk 1]
```diff
@@ -41,6 +41,34 @@ func Platform() (*baremetal.Platform, error) {
...
 	}
 
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "External bridge",
+				Help:    "External bridge is used for external communication.",
+				Default: baremetaldefaults.ExternalBridge,
+			},
+			Validate: survey.ComposeValidators(survey.Required, uriValidator),
```
Comment (russellb) [inline pkg/asset/installconfig/baremetal/baremetal.go]: The URI validator doesn't seem appropriate for the name of a bridge.

We could leave it unvalidated, or we could try to validate that a bridge with that name exists.  If you do that, it's probably not much more work to populate the list of bridges as choices for the interactive text UI.

If you want to leave it unvalidated for this PR, how about open a bug for adding validation later.

**Review by russellb — CHANGES_REQUESTED** [review]

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/684/

File: pkg/asset/installconfig/baremetal/baremetal.go (hunk 1 above)
Comment (stbenjam) [reply pkg/asset/installconfig/baremetal/baremetal.go]: Good catch, you caught me copy and pasting. I've added a validation that checks if the interface exists. It cannot tell us if it's a bridge, without making a more lower level system call but this is at least something.

**Comment by derekhiggins** [comment]
Build ABORTED, see build http://10.8.144.11:8080/job/dev-tools/685/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/686/

**Review by russellb — APPROVED** [review]
I haven't tested this myself, but the changes look good to me.

I'd recommend squashing these 3 commits back down into 1 before merging, though.

**Pull request merged**